lib/streamlit/elements/plotly_chart.py

```diff
@@ -225,11 +225,8 @@ def plotly_chart(
                 check_session_state_rules,
             )
 
-            if current_form_id(self.dg):
-                # TODO(willhuang1997): double check the message of this
-                raise StreamlitAPIException(
-                    "st.plotly_chart with on_select enabled cannot be used inside forms!"
-                )
+            if is_select_enabled and current_form_id(self.dg):
```

Comment:
nit: I don't think you need the `and current_form_id(self.dg)` check here since current_form_id is returning an empty string anyways if its not in a form. Thats how all other widgets are using that. 

```suggestion
            if is_select_enabled:
```